include/dali/core/gpu_utils.h

```diff
@@ -16,17 +16,27 @@
 #define DALI_CORE_GPU_UTILS_H_
 
 #include <cuda_runtime.h>
+#include "dali/core/cuda_utils.h"
 
 namespace dali {
 
 class DeviceGuard {
  public:
+  DeviceGuard() {
+    cudaGetDevice(&original_device_);
+  }
   explicit DeviceGuard(int new_device) {
     cudaGetDevice(&original_device_);
```

Comment:
Why not `CUDA_CALL` on `cudaGetDevice`?

Reply:
And whats the point of having it defined in:
`include/dali/core/gpu_utils.h` and
`include/dali/core/device_guard.h`?